backend/src/org/commcare/session/CommCareSession.java

```diff
@@ -303,8 +303,11 @@ private String getDataNeededByAllEntries(Vector<Entry> entries) {
      * an entry on the stack
      */
     public SessionDatum getNeededDatum() {
-        Entry entry = getEntriesForCommand(getCommand()).elementAt(0);
```

Comment:
Exactly, that is why https://github.com/dimagi/commcare-core/blob/master/backend/src/org/commcare/session/RemoteQuerySessionManager.java#L49-L51 was not hit, because it used to trigger an NPE and now it was triggering a different exception.